Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -692,24 +701,62 @@ public void refreshControls()
     if (mPreferences.getBoolean("showInputOverlay", true))
     {
       // Add all the enabled overlay items back to the HashSet.
-      if (EmulationActivity.isGameCubeGame() || mPreferences.getInt("wiiController", 3) == 0)
-      {
-        addGameCubeOverlayControls(orientation);
-      }
-      else if (mPreferences.getInt("wiiController", 3) == 4)
+      if (EmulationActivity.isGameCubeGame())
       {
-        addClassicOverlayControls(orientation);
+        switch (NativeLibrary
+                .GetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_CORE,
+                        SettingsFile.KEY_GCPAD_PLAYER_1, DISABLED_GAMECUBE_CONTROLLER))
+        {
+          case DISABLED_GAMECUBE_CONTROLLER:
+            if (mIsFirstRun)
+            {
+              Toast.makeText(getContext(), R.string.disabled_gc_overlay_notice, Toast.LENGTH_SHORT)
+                      .show();
+            }
+            break;
+
+          case EMULATED_GAMECUBE_CONTROLLER:
+            addGameCubeOverlayControls(orientation);
+            break;
+
+          case GAMECUBE_ADAPTER:
+            break;
+
+          default:
+            throw new IllegalArgumentException("Invalid SIDevice0 value");
```

Comment:
This should not be an exception. There are other values than the three handled ones that are valid, even if the overlay doesn't support them.